### PR TITLE
Predict/gas error handling

### PR DESCRIPTION
**Pull request by jpeone**
Added some error handling in the predict/gas endpoint, related helper functions, and rate limited my calls to the mapbox API.

### CODE REVIEW

File: project/app/api/predict.py [hunk 1]
```diff
@@ -149,35 +179,87 @@ async def predict_gas(item: GasItem):
...
+            # precise, but this works too.
+            # Example: Ottawa CA (-75.995000, 45.424721)
+            regional_rate = region_gas_predictions(region, month, day, year)
+        except:
+            detail = 'At least one coordinate lays outside the contiguous USA'
+            raise HTTPException(status_code = 422, detail = detail)
+        total += (miles / mpg) * regional_rate
 
+    resp['total'] = round(total, 2)
+    return resp
```
Comment (beccacauthorn) [inline project/app/api/predict.py:203]: Just some clarification here - originally predict_gas was returning a number and now it is returning a dictionary. We just have to make sure that whenever this function is called this should be taken into account.

**Review by beccacauthorn — APPROVED** [review]
Looks good to me, error handling looks awesome, a lot more robust.

File: project/app/api/predict.py (hunk 1 above)
Comment (jpeone) [reply project/app/api/predict.py:203]: Good call. Uvicorn will convert the dict to a json object, so the user will actually get a json object back.

I did a little reading on the weekend, and I guess stylistically a single value json object is preferable to just a float or an int.  Docus page should help inform that usage.

Comment (beccacauthorn) [reply project/app/api/predict.py:203]: Oh gotcha, that makes sense. Thanks for clarifying.

**Pull request merged**